otherlibs/systhreads/st_stubs.c

```diff
@@ -554,7 +554,7 @@ CAMLprim value caml_mutex_try_lock(value wrapper)           /* ML */
 {
   st_mutex mut = Mutex_val(wrapper);
   int retcode = st_mutex_trylock(mut);
-  if (retcode == 1) return Val_false;
+  if (retcode == 0) return Val_false;
```

Comment:
Is it better do something a bit clearer and use a define for the return code like:
https://github.com/ocaml/ocaml/blob/4.10/otherlibs/systhreads/st_stubs.c#L846